### ISSUE TITLE
Create A and AAAA for new project

### ISSUE THREAD

**Issue by benhylau** [hyphacoop/distributed-press-organizing#34]
🎟️ **Re-ticketed from:** #  
📅 **Due date:** end-Feb  
🎯 **Success criteria:** Create all A and AAAA records for a new project.  

## Task Summary

Right now, when we add a new project we have to manually add the domain to Digital Ocean, and create A and AAAA records for the `<project.domain>` and `<api.project.domain>`. This needs to be automated in the API.

There should also be a way to enable/disable WWW, IPFS, and Hypercore publishing in project config. This would affect which records get created.

At the moment we only create the TXT records for DWeb using [this code](https://github.com/hyphacoop/api.distributed.press/blob/19d95344e8755656926ab54a38d566ebceff3f0d/pinning-services/index.js#L194). It should be pretty straightforward to extend this to A and AAAA.

## To Do

- [x] Add enable/disable publishing protocol to project JSON schema
- [x] Update documentation with new JSON schema details for protocol
- [x] Read new protocol settings and apply logic to deployment
- [x] Add HTTP DNS entries 
- [x] Delete DNS entries not enabled
- [x] Create entry in Digital Ocean for domain if it does not exist
- [x] Update [staging script](https://github.com/hyphacoop/hypha.coop/blob/staging/_scripts/config-dweb.json) for hypha disabling http.

**Comment by YurkoWasHere** [comment]
> There should also be a way to enable/disable WWW, IPFS, and Hypercore publishing in project config. This would affect which records get created.

Possible `project.json` config change

```
{
  "domain": "AAAA.BBBB.CCC",
  "protocol": [
    "clear",
    "ipfs",
    "dat"
  ]
}
```
or more detailed with ability to add option in the future.

```
{
  "domain": "AAAA.BBBB.CCC",
  "protocol": {
    "clear":{
        "enable":"true"
    }
    "ipfs":{
        "enable":"true"
    }
    "dat":{
        "enable":"true"
    }
  }
}
```

--

- `protocol`: configurations for specific target protocols
  - `http`: configuration specific to the `http` protocol
    - `enable`: boolean if protocol will be used
  - `ipfs`: configuration specific to the `ipfs` protocol
    - `enable`: boolean if protocol will be used
  - `dat`: configuration specific to the `dat` protocol
    - `enable`: boolean if protocol will be used

**Comment by YurkoWasHere** [comment]
Do we want the ability for custom ipv4 addresses (ie server other then ours)

**Comment by benhylau** [comment]
How's this?
```
{
  "domain": "staging.compost.digital",
  "publication": {
    "protocol": {
      "http": {
        "enable": true
      },
      "ipfs": {
        "enable": true
      },
      "hypercore": {
        "enable": true
      }
    },
    "dns": [
      {
        "type": "A",
        "name": "@",
        "data": "198.50.215.6"
      },
      {
        "type": "AAAA",
        "name": "@",
        "data": "2607:5300:203:4014:5074:f2ff:feb1:a87f"
      }
    ]
  }
}
```

**Comment by YurkoWasHere** [comment]
Well cut out the DNS as it really is out of scope of the requirements

How's this?
```
{
  "domain": "staging.compost.digital",
  "publication": {
    "protocol": {
      "http": {
        "enable": true
      },
      "ipfs": {
        "enable": true
      },
      "hypercore": {
        "enable": true
      }
    }   
  }
}
```
#README text

- `publication`: configuration entries that control publications
  - `protocol`: configurations for specific target protocols
    - `http`: configuration specific to the `HTTP` protocol
      - `enable`: boolean if protocol will be used, default value is `true`
    - `ipfs`: configuration specific to the `IPFS` protocol
      - `enable`: boolean if protocol will be used, default value is `true`
    - `hypercore`: configuration specific to the `HyperCore` protocol
      - `enable`: boolean if protocol will be used, default value is `true`

**Comment by benhylau** [comment]
- [ ] Can you add `false` to http here once this is merged? Thanks! https://github.com/hyphacoop/hypha.coop/blob/staging/_scripts/config-dweb.json

**Comment by YurkoWasHere** [comment]
I was thinking about this last night.  If I'm removing entries when "disabled" i think i need to also add a `"disableDnsUpdate": true` is it does not remove the @ record on disable.
